### PR TITLE
Sigmoid aprox with tanh + fix sigmoid

### LINKED ISSUE #3131: Sigmoid in SMPC failing for some shapes
**Describe the bug**
 Examples to illustrate:

Working
```python
a = torch.Tensor([1,2,3])
a = a.fix_precision().share(bob, alice, crypto_provider = crypto_provider, requires_grad = True)
torch.nn.Sigmoid()(a)
```
Not working
```python
a = torch.Tensor([[1,2,3],[4,5,6]])
a = a.fix_precision().share(bob, alice, crypto_provider = crypto_provider, requires_grad = True)
torch.nn.Sigmoid()(a)
```
Error: `AssertionError: Must be batches of square matrices`

(same with `requires_grad=False`)

The error seems linked to tensor shape

### PR DESCRIPTION
**Pull request by gmuraru**
Fixes #3131 
Also adds the approximation for sigmoid using the tanh.

### CODE REVIEW

File: syft/frameworks/torch/tensors/interpreters/precision.py [hunk 1]
```diff
@@ -498,12 +498,8 @@ def sigmoid(self, method="exp"):
...
 
         if method == "exp":
-            # Inverse can only be used on matrices
-            if len(self.shape) == 1:
-                one = self * 0 + 1
-                result = one / (1 + (self * -1).exp())
-            else:
-                result = (1 + (self * -1).exp()).inverse()
+            one = self * 0 + 1
+            result = one / (1 + (self * -1).exp())
```
Comment (LaRiffle) [inline syft/frameworks/torch/tensors/interpreters/precision.py]: So basically you're suggesting .inverse() doesn't work ? :(
(The benefit of .inverse is that it's supposed to be faster)

Comment (LaRiffle) [reply syft/frameworks/torch/tensors/interpreters/precision.py]: But yeah you're right it's failiing apparently sometimes, I was discussing of this two weeks ago on the call with André

File: syft/frameworks/torch/tensors/interpreters/precision.py [hunk 2]
```diff
@@ -560,6 +607,7 @@ def _tanh_chebyshev(tensor, maxval: int = 6, terms: int = 32):
           where c_i is the ith Chebyshev series coefficient and P_i is ith polynomial.
         The approximation is truncated to +/-1 outside [-maxval, maxval].
         Args:
+            tensor (tensor): values where the tanh needs to be approximated
```
Comment (LaRiffle) [inline syft/frameworks/torch/tensors/interpreters/precision.py]: typo this should be here :)

File: test/torch/tensors/test_precision.py [hunk 3]
```diff
@@ -449,6 +449,7 @@ def test_torch_sigmoid_approx(workers):
     alice, bob, james = workers["alice"], workers["bob"], workers["james"]
 
     fix_prec_tolerance_by_method = {
+        "chebyshev": {3: 6 / 100, 4: 1 / 100, 5: 1 / 100},
```
Comment (LaRiffle) [inline test/torch/tensors/test_precision.py]: To improve test speed could we remove the case precision=5 everywhere where it occurs? From what I see =4 is already compelling enough :)

File: syft/frameworks/torch/tensors/interpreters/precision.py [hunk 4]
```diff
@@ -482,45 +486,88 @@ def exp(self, iterations=8):
...
+    def sign(self):
+        return (self > 0) + (self < 0) * (-1)
+
+    @staticmethod
+    def _sigmoid_exp(tensor):
         """
-        Approximates the sigmoid function
+        Compute the sigmoid using the exp approximation
+        sigmoid(x) = 1 / (1 + exp(-x))
+        sigmoid(x) = (sigmoid(|x|) - 0.5) * sign(x) + 0.5
```
Comment (LaRiffle) [inline syft/frameworks/torch/tensors/interpreters/precision.py]: Detail a bit the method for people which don't know about numerical stability etc

File: syft/frameworks/torch/tensors/interpreters/precision.py [hunk 5]
```diff
@@ -449,6 +449,10 @@ def matmul(self, *args, **kwargs):
     __matmul__ = matmul
     mm = matmul
 
+    def reciprocal(self):
+        ones = self * 0 + 1
+        return ones / self
```
Comment (LaRiffle) [inline syft/frameworks/torch/tensors/interpreters/precision.py]: Let's keep this like this, until I come back to trying my approx inverse trick

File: syft/frameworks/torch/tensors/interpreters/precision.py (hunk 2 above)
Comment (gmuraru) [reply syft/frameworks/torch/tensors/interpreters/precision.py]: Hmm..this is for ```_tanh_chebyshev``` (forgot to put it in the previous PR)

**Pull request merged**